### PR TITLE
Constrain plugins to normal range when assigning ride prices

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

File: src/openrct2/scripting/bindings/ride/ScRide.cpp [hunk 1]
```diff
@@ -325,7 +325,7 @@ namespace OpenRCT2::Scripting
             auto numPrices = std::min(value.size(), ride->GetNumPrices());
             for (size_t i = 0; i < numPrices; i++)
             {
-                ride->price[i] = static_cast<money64>(value[i]);
+                ride->price[i] = std::clamp<money64>(value[i], 0.00_GBP, 20.00_GBP);
```
Comment (Gymnasiast) [inline src/openrct2/scripting/bindings/ride/ScRide.cpp:328]: There should be a constant for the maximum price. (If there isn’t, there should be :-P)

File: src/openrct2-ui/windows/Ride.cpp [hunk 2]
```diff
@@ -6206,7 +6206,7 @@ static_assert(std::size(RatingNames) == 6);
                 return;
             }
 
-            price = std::clamp(price, 0.00_GBP, 20.00_GBP);
+            price = std::clamp(price, kRideMinPrice, kRideMaxPrice);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Ride.cpp:6209]: It’s good that the window does a bounds check, but the action should do so as well, I’d say. Would you agree?

Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/Ride.cpp:6209]: How that you mention it, I think the action should fail outright if anything out of bounds is presented, indeed.

**Review by Gymnasiast — APPROVED** [review]

**Comment by Sadret** [comment]
This was merged really quickly considering it affects existing plugins. Luckily not a big deal, red error box ingame and error on the command line, but no exception thrown causing the plugin to stop.

**Comment by AaronVanGeffen** [comment]
Sorry to hear this adversely affected a plugin of yours. We were more concerned with people actively exploiting setting negative prices, and didn't consider there might be someone using it to set prices above 20GBP... 😅

**Comment by cheweytoo** [comment]
Interesting! I expected this change to have a much steeper impact on park finances than it actually has.

From a park with 19 rides that previously had prices above 20 set via price manager:
no_more_fleecing
(change happened on 4th August)

It seems that the reduced price is at least partially compensated by the fact that now more guests can afford the rides.

I guess I'm fine with the change then 😎.

**Pull request merged**